### PR TITLE
feat(integrations-mapping): Add integration mapping GraphQL mutations

### PR DESCRIPTION
**Pull request by ivannovosad**
## Roadmap Task

👉  https://getlago.canny.io/feature-requests/p/integration-with-netsuite

## Context

Currently Lago does not support accounting integrations.

## Description

This Pull request adds GraphQL mutations for netsuite integration and collection mappings.

### CODE REVIEW

File: app/graphql/mutations/integration_collection_mappings/destroy.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+      field :id, ID, null: true
+
+      def resolve(id:)
+        integration_collection_mapping = ::IntegrationCollectionMappings::BaseCollectionMapping
+          .joins(:integration)
+          .where(id:)
+          .where(integration: { organization: current_organization }).first
+
+        return not_found_error(resource: 'integration_collection_mapping') unless integration_collection_mapping
```
Comment (vincent-pochet) [inline app/graphql/mutations/integration_collection_mappings/destroy.rb:22]: It looks like this logic is already handled in the destroy service. Any reason for having it here as well?

File: app/graphql/mutations/integration_mappings/destroy.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      field :id, ID, null: true
+
+      def resolve(id:)
+        integration_mapping = ::IntegrationMappings::BaseMapping
+          .joins(:integration)
+          .where(id:)
+          .where(integration: { organization: current_organization }).first
+
+        return not_found_error(resource: 'integration_mapping') unless integration_mapping
```
Comment (vincent-pochet) [inline app/graphql/mutations/integration_mappings/destroy.rb:22]: Same comment regarding the check for presence in the destroy service

**Review by vincent-pochet — APPROVED** [review]

File: app/graphql/mutations/integration_collection_mappings/netsuite/create.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      class Create < BaseMutation
+        include AuthenticableApiUser
+        include RequiredOrganization
+
+        graphql_name 'CreateNetsuiteIntegrationCollectionMapping'
+        description 'Create Netsuite integration collection mapping'
+
+        input_object_class Types::IntegrationCollectionMappings::Netsuite::CreateInput
+
+        type Types::IntegrationCollectionMappings::Netsuite::Object
```
Comment (lovrocolic) [inline app/graphql/mutations/integration_collection_mappings/netsuite/create.rb:15]: I am not sure why do we have `netsuite` specific object to return. IMO we should use generic object which can be used for other integrations as well

File: app/graphql/mutations/integration_collection_mappings/netsuite/update.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      class Update < BaseMutation
+        include AuthenticableApiUser
+        include RequiredOrganization
+
+        graphql_name 'UpdateNetsuiteIntegrationCollectionMapping'
+        description 'Update Netsuite integration mapping'
+
+        input_object_class Types::IntegrationCollectionMappings::Netsuite::UpdateInput
+
+        type Types::IntegrationCollectionMappings::Netsuite::Object
```
Comment (lovrocolic) [inline app/graphql/mutations/integration_collection_mappings/netsuite/update.rb:15]: Same comment as above

**Pull request merged**